### PR TITLE
feat(elastic): allow kwargs to be pass at extend

### PR DESCRIPTION
**Pull request by alphinside**
Goals:

- Enable elastic search `parallel_bulk` parameter passing from docarray `extend` method
- [x] check and update documentation, if required. See [guide](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md#documentation-guidelines)

### CODE REVIEW

File: docarray/array/storage/elastic/backend.py [hunk 1]
```diff
@@ -81,6 +81,19 @@ def _init_storage(
         self._index_name_offset2id = 'offset2id__' + config.index_name
         self._config = config
 
+        # Only accept several params for es parallel bulk operation
+        accepted_bulk_config = [
+            "thread_count",
```
Comment (JoanFM) [inline docarray/array/storage/elastic/backend.py:86]: please use single quote

File: docarray/array/storage/elastic/backend.py [hunk 2]
```diff
@@ -40,7 +40,7 @@ class ElasticConfig:
     es_config: Dict[str, Any] = field(default_factory=dict)
     index_text: bool = False
     tag_indices: List[str] = field(default_factory=list)
-    batch_size: int = 64
+    bulk_config: Dict[str, Any] = field(default_factory=dict)
```
Comment (JoanFM) [inline docarray/array/storage/elastic/backend.py:43]: is there a way this can be made non breaking change? can we detect if batch size is added?

**Review by JoanFM — CHANGES_REQUESTED** [review]
I am not convinced by passing this in the config? maybe we should just pass *args and **kwargs?

**Comment by JoanFM** [comment]
> I am not convinced by passing this in the config? maybe we should just pass *args and **kwargs?

@samsja @JohannesMessner any opinion?

File: docarray/array/storage/elastic/backend.py (hunk 1 above)
Comment (alphinside) [reply docarray/array/storage/elastic/backend.py]: noted

File: docarray/array/storage/elastic/backend.py (hunk 2 above)
Comment (alphinside) [reply docarray/array/storage/elastic/backend.py:43]: sure, will add it

Comment (alphinside) [reply docarray/array/storage/elastic/backend.py:44]: done, if the bulk config is provided, the configuration from bulk config will take priority

**Comment by samsja** [comment]
> > I am not convinced by passing this in the config? maybe we should just pass *args and **kwargs?
> 
> @samsja @JohannesMessner any opinion?

I believe we should stick to pass **kwars, *args for now. THe only issue will not be able to pass kwargs when doing operation via python operator

File: docarray/array/storage/elastic/backend.py [hunk 3]
```diff
@@ -159,10 +159,32 @@ def _build_client(self):
...
+    def _send_requests(self, request, **kwargs) -> List[Dict]:
         """Send bulk request to Elastic and gather the successful info"""
+        bulk_config = {}
+
+        # for backward compatibility
+        if 'chunk_size' not in kwargs:
+            bulk_config['chunk_size'] = self._config.batch_size
+
+        # Only accept several params for es parallel bulk operation
+        accepted_config = [
```
Comment (JoanFM) [inline docarray/array/storage/elastic/backend.py:171]: this is not needed. Leave the freedom to user. We do not want to mantain this for every backend. We let the power to the user. We should add the corresponding docstring and that is.

File: docarray/array/storage/elastic/backend.py [hunk 4]
```diff
@@ -159,10 +159,32 @@ def _build_client(self):
         client.indices.refresh(index=self._config.index_name)
         return client
 
-    def _send_requests(self, request) -> List[Dict]:
+    def _send_requests(self, request, **kwargs) -> List[Dict]:
         """Send bulk request to Elastic and gather the successful info"""
+        bulk_config = {}
```
Comment (JoanFM) [inline docarray/array/storage/elastic/backend.py:164]: do not call bulk_config. From our side it should be simply calling kwargs, would not add any semantic meaning

File: docs/advanced/document-store/elasticsearch.md [hunk 5]
```diff
@@ -92,41 +92,77 @@ da2.summary()
...
+
 [0.14890289 0.3168339  0.03050802 0.06785086 0.94719299 0.32490566
  ...]
 ```
 
 Other functions behave the same as in-memory DocumentArray.
 
+### Bulk request configuration
+
+You can configure how bulk requests is being sent to Elasticsearch when adding documents by adding additional `kwargs` when adding documents. Bulk operation is used for several index operations; like (`index` or `delete`). See [the official Documentation](https://elasticsearch-py.readthedocs.io/en/v8.3.3/helpers.html) for more details. Notes that only the following parameters are accepted, the others will be ignored
```
Comment (JoanFM) [inline docs/advanced/document-store/elasticsearch.md:124]: This is not configuration. This is customization perhaps

Comment (JoanFM) [reply docs/advanced/document-store/elasticsearch.md:124]: Remove reference to what is accepted and what not. User's responsability

**Review by JoanFM — CHANGES_REQUESTED** [review]
Some comments

File: docarray/array/storage/elastic/backend.py (hunk 3 above)
Comment (alphinside) [reply docarray/array/storage/elastic/backend.py:171]: noted

File: docarray/array/storage/elastic/backend.py (hunk 4 above)
Comment (alphinside) [reply docarray/array/storage/elastic/backend.py:164]: noted

File: docs/advanced/document-store/elasticsearch.md (hunk 5 above)
Comment (alphinside) [reply docs/advanced/document-store/elasticsearch.md:124]: noted

**Review by JohannesMessner — APPROVED** [review]
LGTM, thanks for the great contribution!

**Review by JoanFM — APPROVED** [review]

**Pull request merged**